src/core/opcode.c

```diff
@@ -69,32 +84,58 @@ void chip8_opcode_7XNN(chip8_cpu *cpu)
 
 void chip8_opcode_8XY0(chip8_cpu *cpu)
 {
-	/* TODO */
+	uint8_t x = (cpu->opcode & 0x0F00) >> 8;
+	uint8_t y = (cpu->opcode & 0x00F0) >> 4;
+	cpu->v[x] = cpu->v[y];
 }
 
 void chip8_opcode_8XY1(chip8_cpu *cpu)
 {
-	/* TODO */
+	uint8_t x = (cpu->opcode & 0x0F00) >> 8;
+	uint8_t y = (cpu->opcode & 0x00F0) >> 4;
+	cpu->v[x] = cpu->v[x] | cpu->v[y];
 }
 
 void chip8_opcode_8XY2(chip8_cpu *cpu)
 {
-	/* TODO */
+	uint8_t x = (cpu->opcode & 0x0F00) >> 8;
+	uint8_t y = (cpu->opcode & 0x00F0) >> 4;
+	cpu->v[x] = cpu->v[x] & cpu->v[y];
 }
 
 void chip8_opcode_8XY3(chip8_cpu *cpu)
 {
-	/* TODO */
+	uint8_t x = (cpu->opcode & 0x0F00) >> 8;
+	uint8_t y = (cpu->opcode & 0x00F0) >> 4;
+	cpu->v[x] = cpu->v[x] ^ cpu->v[y];
 }
 
 void chip8_opcode_8XY4(chip8_cpu *cpu)
 {
-	/* TODO */
+	uint16_t tmp = 0;
+	uint8_t x = (cpu->opcode & 0x0F00) >> 8;
+	uint8_t y = (cpu->opcode & 0x00F0) >> 4;
+	tmp = cpu->v[x] + cpu->v[y];
+        cpu->v[0x0f] = false;
+        if (tmp > 0xff)
+        {
+            cpu->v[0x0f] = true;
+        }
+
+        cpu->v[x] = tmp;
 }
 
 void chip8_opcode_8XY5(chip8_cpu *cpu)
 {
-	/* TODO */
+	uint16_t tmp = 0;
+	uint8_t x = (cpu->opcode & 0x0F00) >> 8;
+	uint8_t y = (cpu->opcode & 0x00F0) >> 4;
+	cpu->v[0x0f] = false;
+        if (cpu->v[x] > cpu->v[y])
+        {
+            cpu->v[0x0f] = true;
```

Comment:
According to the CHIP-8 specification the VF register must be set to false in case of borrow, i.e., underflow.